icu4j/main/classes/core/src/com/ibm/icu/impl/PluralRulesLoader.java

```diff
@@ -28,7 +29,8 @@
  * Loader for plural rules data.
  */
 public class PluralRulesLoader extends PluralRules.Factory {
-    private final Map<String, PluralRules> rulesIdToRules;
+    // Key is locale + "/" + type
```

Comment:
Previously, the key was the "rule set".  Locales that share the same rule set used the same PluralRules instance.  As written, my change will have the result of making the cache substantially larger.  Since plural ranges are also defined by their set, I can improve this a bit by caching on the pair of plural rule set and plural range set.  Alternatively, I can cache the parsed plural rules separately from the parsed plural ranges.

Reply:
If you have time, please either cache range rules separately or build a key from both rule-set strings.